SimCalorimetry/HGCalSimProducers/src/HGCDigitizerBase.cc

```diff
@@ -36,6 +37,17 @@ HGCDigitizerBase<DFr>::HGCDigitizerBase(const edm::ParameterSet& ps) {
   edm::ParameterSet feCfg = myCfg_.getParameter<edm::ParameterSet>("feCfg");
   myFEelectronics_ = std::unique_ptr<HGCFEElectronics<DFr>>(new HGCFEElectronics<DFr>(feCfg));
   myFEelectronics_->SetNoiseValues(noise_fC_);
+  myFEelectronicstoaMode_ = myFEelectronics_->toaMode();
+  Dim_ = noise_fC_.size();
+}
+
+template <class DFr>
+void HGCDigitizerBase<DFr>::GenGaussianNoise(const double NoiseMean, const double NoiseStd) {
+  for (size_t i = 0; i < NoiseArrayLength_; i++) {
+    for (int j = 0; j < samplesize_; j++) {
+      GaussianNoiseArray_[i][j] = CLHEP::RandGaussQ::shoot(NoiseMean, NoiseStd);
```

Comment:
@adas1994 , you need somehow to use randgauss from the next method, here is not clear where from random engine is coming - it may be initialized or not.